### ISSUE TITLE
mone in generator

### ISSUE THREAD

**Issue by udion** [jalola/improved-wgan-pytorch#1]
Hi,
In the gan training code, since you have already used `gen_cost.backward(mone)` for generator, why are you still doing `gen_cost = -gen_cost`

i.e, should it not be 
* just `gen_cost.backward(mone)`
or
*  `gen_cost.backward()` and then `gen_cost = gen_cost*-1`
Please correct me if I am wrong.

**Comment by jalola** [comment]
Hi,

The last `gen_cost = -gen_cost` does not affect the gradients or the optimizer_g.step() because after backward(), gradients are calculated. It only affects the graph when we draw the gen_cost (It just flips vertically when you're putting - or not putting -)

What matters here is:
`gen_cost.backward(mone)` and 
`disc_cost = disc_fake - disc_real + gradient_penalty`
`disc_cost.backward()`

After `optimizer.step()`, G tries to maximize gen_cost and D tries to minimize disc_fake(kind of gen_cost). So there is a fight between G and D.

**Comment by udion** [comment]
yes thank you.
Tell me though are these 2 equivalent:
*  `gen_cost = -gen_cost` and then `gen_cost.backward()`
* no need to invert `gen_cost` before calling `backward` and instead pass `mone` in backward (i.e what you did)

**Comment by jalola** [comment]
thanks too, 

That's true

`gen_cost = -gen_cost` 
`gen_cost.backward()`

would result the same.

**Comment by Nikasa1889** [comment]
The issue is solved.
